ext/node/polyfills/internal_binding/stream_wrap.ts

```diff
@@ -311,21 +311,37 @@ export class LibuvStreamWrap extends HandleWrap {
 
   /** Internal method for reading from the attached stream. */
   async #read() {
-    const isOwnedBuf = bufLocked;
-    let buf = bufLocked ? new Uint8Array(SUGGESTED_SIZE) : BUF;
-    bufLocked = true;
+    // Lock safety: We must hold this lock until we are certain that buf is no longer used
+    // This setup code is a little verbose, but we need to be careful about buffer management
+    let buf, locked = false;
+    if (bufLocked) {
+      // Already locked, allocate
+      buf = new Uint8Array(SUGGESTED_SIZE);
+    } else {
+      // Not locked, take the buffer + lock
+      buf = BUF;
+      locked = bufLocked = true;
```

Comment:
Wouldn't the code be much simpler if you always allocate `buf = new Uint8Array(SUGGESTED_SIZE);`? `locked,  bufLocked` will be deemed useless.

Is the current shared buffer motivated only by a performance gain? @mmastrac